src/test/css/folding.test.ts

```diff
@@ -180,6 +180,19 @@ suite('CSS Folding - Nested', () => {
 	});
 });
 
+suite('CSS Folding - Regions', () => {
+	test('Simple region with comment', () => {
+		const input = [
+			/*0*/'/* #region */',
+			/*1*/'& .bar {',
+			/*2*/'  color: red;',
+			/*3*/'}',
+			/*4*/'/* #endregion */'
+		];
+		assertRanges(input, [r(0, 4, 'region'), r(1, 2)]);
+	});
+});
+
```

Comment:
please add some tests with incorrect nesting, e.g. the
```
/* #region */
& .bar {
/* #endregion  */
}
```
The region should win...

Reply:
Regions can not intersect. Maybe look at this example:
```
/* #region */
.bar {
/* #endregion  */
// hello
}
```
[0,2] and [1,3] would be illegal.
-> once you see the /* #endregion  */, discard all open ranges until you see the open /* #region */.

Reply:
OK, will open a new issue to track this.

Reply:
I created #85.